src/with-filter.ts

```diff
@@ -1,5 +1,5 @@
 export type FilterFn<TSource = any, TArgs = any, TContext = any> = (rootValue?: TSource, args?: TArgs, context?: TContext, info?: any) => boolean | Promise<boolean>;
-export type ResolverFn<TSource = any, TArgs = any, TContext = any> = (rootValue?: TSource, args?: TArgs, context?: TContext, info?: any) => AsyncIterator<any> | Promise<AsyncIterator<any>>;
+export type ResolverFn<TSource = any, TArgs = any, TContext = any> = (rootValue?: TSource, args?: TArgs, context?: TContext, info?: any) => AsyncIterableIterator<any> | Promise<AsyncIterableIterator<any>>;
```

Comment:
AsyncIterableIterator honestly is a kind of strange type — I guess it basically means a thing that is both AsyncIterator and AsyncIterable, even though they are distinct concepts? It feels like it might be nicer to allow you to pass in a thing that is *either* an iterator or an iterable, not something that has to be *both*. But I guess this is a path this code has already gone.

This is a major version bump, right? Because this definitely changes the requirements on people calling `withFilter`.